docs/docs/development/hardware-integration/pinctrl.mdx

```diff
@@ -0,0 +1,345 @@
+
+Note that not all peripherals require all pins to be defined - for example, there are plenty of SPI devices which do not require either the MISO (Main In, Sub Out) or the MOSI (Main Out, Sub In) pin.
+
+## Pin Control File
+
+Pin control consists of two parts:
+
+1. Defining and grouping the pins together for a driver
+2. Defining a driver node and assigning pin groups to it
+
+It is standard to do this in two separate files. Create a file called `<my-board>-pinctrl.dtsi`, which will be used for the first part. This file will later be imported into your board's `.dts`. To avoid namespace collisions, if you are configuring pin control for a [shield](#boards-shields-and-modules), then you should write the contents of both parts into the `.overlay` file directly.
+
+### MCU/SoC Pinctrl Bindings Files
+
+The specifics of pin control for a particular MCU/SoC are presented via a [devicetree bindings](https://docs.zephyrproject.org/latest/build/dts/bindings-intro.html) file, found under `zephyr/dts/bindings/pinctrl/`.
+
+- The nRF52840 uses the `zephyr/dts/bindings/pinctrl/nordic,nrf-pinctrl.yaml` file.
+- The rp2040 uses the `zephyr/dts/bindings/pinctrl/raspberrypi,pico-pinctrl.yaml` file.
+
+These files often contain useful comments on pin control for their devices, and so can be worth a read in addition to this page.
+
+### MCU/SoC Pinctrl Bindings Headers
+
+MCUs/SoCs will also have pinctrl header files found under `zephyr/include/zephyr/dt-bindings/pinctrl/`. You will want to make sure that the header file corresponding to your MCU/SoC is imported into your board's `.dts`/`.overlay`. For the nRF52840, your board should already be importing a file such as `nordic/nrf52840_qiaa.dtsi`, which includes the pinctrl header via the following inclusions:
+
+```plaintext
+nrf-pinctrl.h -> nrf_common.dtsi -> nrf52840.dtsi -> nrf52840_qiaa.dtsi
+```
+
+The corresponding file for the rp2040 (`rpi_pico/rp2040.dtsi`), however, does not include the pinctrl header. Hence you will need to import said header at the top of your `<my-board>-pinctrl.dtsi` file:
+
+```dts title="<my-board>-pinctrl.dtsi"
+#include <dt-bindings/pinctrl/rpi-pico-rp2040-pinctrl.h>
+```
+
+If you are configuring pin control for a [shield](#boards-shields-and-modules), then this file may already be imported by the board's definition, in which case you shouldn't re-include it here.
+
+:::info
+Some MCUs/SoCs may have their headers located in a Zephyr HAL module. See for example the [NXP HAL module](https://github.com/zephyrproject-rtos/hal_nxp/tree/master/dts/nxp/nxp_imx/rt).
+:::
```

Comment:
Maybe it's better to use the SAMD21 HAL as an example? I think more people will be developing boards using the samd21 rather than the teensy, although admittedly it has some complications around it's SERCOM blocks rather than discrete spi,i2c etc blocks. The imx pinctrl is also rather complex in its own way

Reply:
I have no particular opinion either way, that change has been made.